fastlane_bot/helpers/carbon_trade_splitter.py

```diff
@@ -1,203 +1,71 @@
-import json
```

Comment:
This comment applies to this file as well as the wrap processor file. 

Throughout the library when there is a processor we generally follow the same pattern, notably 


    FooProcessor(...).process(...)


with FooProcessor often, but not necessarily a dataclass, rather than 
    
    process_foo(...)


This is  a question of style and there are arguments for both: notably the latter is more in line with functional programming in that it typically is a pure function (or at least should be...) and therefor can be clearer.

The advantage of the former include

**Namespacing and scope**. To run the processor you often you need constants, or additional functionality eg for pre- and post-processing. Traditionally in Python that was handled using modules, ie modules would contain the constants and functions that were needed. You
then write

    import module as m
    result = m.process(param=m.CONSTANT, ...)
    pretty_result = m.pretty(result)
    
Nowadays you can use classes to achieve the same result

    from module import Processor
    P = Processor(...)
    result = P.process(param=Processor.CONSTANT, ...)
    pretty_result = P.pretty(result) # regular method
    pretty_result = Processor.pretty(result) # class method

The advantage is here that you can import the classes into a higher level module
and you do not lose the namespace

    # helpers/__init__.py
    from .processor import Processor

    # main.py
    from helpers import Processor
    result = Processor(...).process(...)

This code is using the following pattern

    # helpers/__init__.py
    from .processor import process

    # main.py
    from helpers import process
    result = process(...)

This loses the namespace, and if we have constants (now or in the future) then we will either import them from the low level module, or we risk namespace conflicts

**Handling of default parameters**. Often processors will have some parameters that are
changeable, but not usually changed. Now they _can_ be on a config object that can 
be passed, but sometimes this is tedious if those parameters are really usually the same.

In this case you could do either

    FooProcessor(cfg=config, othercfg=<newparam>).process(...)

or even

    class FooProcessor1(FooProcessor):
        OTHERCFG = <newparam>

    FooProcessor1(cfg=config).process(...)



 Even if you think that this is overkill here I would still suggest to wrap the processing functions into a class to (a) be in line with practices mostly employed here, and (b) to allow for future extensibility

Reply:
```
FooProcessor(...).process(...)
```
This results with a `FooProcessor` object which is created just for the sake of calling a function, destroyed immediately thereafter.
Hence, we can and should just declare that function as a global function.
For the purpose of scoping that function within its own module, implementing it in a separate file is sufficient; there is no need to further scope it inside a class within that file, as there is no additional benefit in doing so.

Reply:
that ignores the `dataclass` part; I agree that self should not carry any data not declared as dataclass field (and generally we should avoid passing state via self - self properties should by and large be context and configuration); however -- as dataclass it is perfectly readable